src/coreclr/vm/jitinterface.cpp

```diff
@@ -3086,19 +3089,14 @@ void CEEInfo::ComputeRuntimeLookupForSharedGenericToken(DictionaryEntryKind entr
     // Unless we decide otherwise, just do the lookup via a helper function
     pResult->indirections = CORINFO_USEHELPER;
 
-    // Runtime lookups in inlined contexts are not supported by the runtime for now
-    if (pResolvedToken->tokenContext != METHOD_BEING_COMPILED_CONTEXT())
-    {
-        pResultLookup->lookupKind.runtimeLookupKind = CORINFO_LOOKUP_NOT_SUPPORTED;
-        return;
-    }
+    bool inlinedLookup = pResolvedToken->tokenContext != METHOD_BEING_COMPILED_CONTEXT();
+    MethodDesc* pContextMD = inlinedLookup ? pContainingMD : GetMethodFromContext(pResolvedToken->tokenContext);
```

Comment:
Use `pContainingMD` unconditionally? This should not need to special-case inlinedLookup vs. non-inlinedLookup.

Also, I think you can get rid of `METHOD_BEING_COMPILED_CONTEXT`. It should not be needed, and it is unnecessary complexity with your change.

Reply:
I think it should work, addressed